packages/delegate/src/defaultMergedResolver.ts

```diff
@@ -28,9 +42,307 @@ export function defaultMergedResolver(
     return defaultFieldResolver(parent, args, context, info);
   }
 
+  // If the parent is satisfied for the left over after a nested delegation, try to resolve it
+  if (!parent.hasOwnProperty(responseKey)) {
+    const leftOver = getPlanLeftOverFromParent(parent);
+    // Add this field to the deferred fields
+    if (leftOver) {
+      let missingFieldNodes = leftOver.missingFieldsParentMap.get(parent);
+      if (!missingFieldNodes) {
+        missingFieldNodes = [];
+        leftOver.missingFieldsParentMap.set(parent, missingFieldNodes);
+      }
+      missingFieldNodes.push(
+        ...info.fieldNodes.filter(fieldNode =>
+          leftOver.unproxiableFieldNodes.some(
+            unproxiableFieldNode => unproxiableFieldNode === fieldNode,
+          ),
+        ),
+      );
+      let missingDeferredFields = leftOver.missingFieldsParentDeferredMap.get(parent);
+      if (!missingDeferredFields) {
+        missingDeferredFields = new Map();
+        leftOver.missingFieldsParentDeferredMap.set(parent, missingDeferredFields);
+      }
+      const deferred = createDeferred();
+      missingDeferredFields.set(responseKey, deferred);
+      return deferred.promise;
+    }
+    return undefined;
+  }
+  return handleResult(parent, responseKey, context, info);
+}
+
+function handleResult(
+  parent: ExternalObject,
+  responseKey: string,
+  context: Record<string, any>,
```

Comment:
```suggestion
  context: Record<string, unknown>,
```

Reply:
Changed it anyways similar to https://github.com/ardatan/graphql-tools/pull/6150#discussion_r1593781816